### PR TITLE
Add conductor-tenancy-sanitizer

### PR DESCRIPTION
**Pull request by tomasol**
Summary: Add proxy layer (wip) that will sanitize conductor
requests/responses and handle multi tenancy (prefixing,
validation,search filtering...)

This project will live either as a separate container
or it might be integrated into conductor container.
This commit is just a template project, functionality
will be added gradually in next PRs.

### CODE REVIEW

File: symphony/integration/conductor-tenancy-sanitizer/server/src/main/main10.iml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (KenG98) [inline symphony/integration/conductor-tenancy-sanitizer/server/src/main/main10.iml]: I think this file was created by IntelliJ and shouldn't be committed?

**Review by KenG98 — COMMENTED** [review]
Couple notes:

1. Please move these changes from the `integration` folder to `/devmand/workflow/`. You'll need to create the `workflow` folder. We'll move this into the relevant hub folder soon, we just haven't open sourced it yet. `integration` is more for orchestration (docker, make, env, etc).

Couple questions:

1. I see you're writing in Java. Am I correct in assuming the conductor proxy will be in Java, and the Workflow Builder backend will be in Node.js?
2. The proxy will eventually run in a docker container, right?
3. The project is configured to use Java 8 - should we use Java 11, the latest LTS version? I believe Java 8 is EOL this year.

This looks good 👍 Thank you!

File: symphony/integration/conductor-tenancy-sanitizer/server/src/main/main10.iml (hunk 1 above)
Comment (tomasol) [reply symphony/integration/conductor-tenancy-sanitizer/server/src/main/main10.iml]: Thanks for catching this, fixed.

**Comment by tomasol** [comment]
I have moved the folder as suggested.

> I see you're writing in Java. Am I correct in assuming the conductor proxy will be in Java, and the Workflow Builder backend will be in Node.js?

Yes. The proxy is much closer to conductor itself, so we will be sharing code and jumping between conductor codebase and proxy more often. Since multi tenancy was already discussed in conductor, we may offer (if you agree) this code as a PR back to conductor. That is why I am using the same build tool and libraries.
> The proxy will eventually run in a docker container, right?

Yes. I know java tends to be more memory hungry, we may create a container where both conductor and proxy run in the same process, or we may go with separate containers.
> The project is configured to use Java 8 - should we use Java 11, the latest LTS version? I believe Java 8 is EOL this year.

In the docker image we may either specify Java 8 that is still supported (backed by Amazon or Red Hat) or jump to Java 11. The code base specifies that source and target should be compatible with Java 8, which is in line with Conductor for the reasons mentioned above.

**Comment by tomasol** [comment]
As per our discussion I am closing this and going to submit a JS version

**Pull request closed without merging**